src/theory/sets/theory_sets_private.cpp

```diff
@@ -286,6 +286,10 @@ void TheorySetsPrivate::fullEffortCheck()
           d_card_enabled = true;
           // register it with the cardinality solver
           d_cardSolver->registerTerm(n);
+          if (d_im.hasSent())
+          {
+            break;
```

Comment:
Why does this break solve the issue,  when there is a  new lemma  or fact?

Reply:
`d_cardSolver->registerTerm` may send a lemma, which then calls Theory::preRegisterTerm, which in turn modifies the equality engine, making the equality engine iterator we are using invalid.

Reply:
Thanks for the clarification.